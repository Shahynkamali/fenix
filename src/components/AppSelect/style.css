.select {
  @apply inline-flex;
}

.select__button {
  @apply mt-1 form-select block w-full pl-3 pr-10 py-2 text-base leading-6 border-gray-300  xs:leading-5 xl:cursor-pointer lg:cursor-pointer;
}

.select__button:focus {
  @apply border-blue-300 shadow-outline-blue outline-none ;
}
