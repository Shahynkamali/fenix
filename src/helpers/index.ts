// eslint-disable-next-line import/prefer-default-export
export enum SORTBYAGE {
  ASCENDING = 'Ascending Age ',
  DESCENDING = 'Descending Age',
}

export enum HAIRCOLORS {
  ALL = 'All Hair Colors',
  BLACK = 'Black',
  RED = 'Red',
  GREEN = 'Green',
  GRAY = 'Gray',
  PINK = 'Pink',
}
