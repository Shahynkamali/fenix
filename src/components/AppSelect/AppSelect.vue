<template>
  <div class="select">
    <label
      :for="label"
      class="sr-only"
    >
      {{ label }}
    </label>
    <select
      :id="label"
      v-model="selectedOption"
      class="select__button"
      v-bind="$attrs"
    >
      <option
        v-for="(option, index) in options"
        :key="index"
        :value="option"
      >
        {{ option }}
      </option>
    </select>
  </div>
</template>

<script lang="ts">
import {
  Component, Vue, Prop, Watch,
} from 'vue-property-decorator';
import './style.css';

@Component({
  name: 'AppSelect',
  inheritAttrs: false,
})
export default class AppSelect extends Vue {
  @Watch('selectedOption')
  emitValueToParent(payload: string) {
    this.$emit('input', payload);
  }

    @Prop({ required: true })
    readonly options!: string[];

    @Prop({ required: false, default: 'select' })
    readonly label!: string;

    @Prop({ required: false })
    readonly value!: string;

    private selectedOption: null | string = null;

    private mounted(): void {
      this.selectedOption = this.value;
    }
}
</script>
