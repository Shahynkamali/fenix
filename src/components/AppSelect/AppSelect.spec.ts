import { mount } from '@vue/test-utils';
import { SORTBYAGE } from '@/helpers';
import AppSelect from './AppSelect.vue';

describe('AppSelect', () => {
  const factory = (propsData?: any) => mount(AppSelect, {
    propsData: {
      ...propsData,
    },
  });

  it('should display the selected option', async () => {
    const optionsProp = {
      options: SORTBYAGE,
    };
    const wrapper = factory(optionsProp);
    const options = wrapper.find('select').findAll('option');
    await options.at(1).setSelected();
    expect(wrapper.find('option').element.innerHTML.trim()).toBe(SORTBYAGE.ASCENDING.trim());
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should accept a array of strings as a prop and set that as options', () => {
    const optionsProp = {
      options: SORTBYAGE,
    };
    const wrapper = factory(optionsProp);
    expect(wrapper.vm.$props.options).toBe(SORTBYAGE);
    expect(wrapper.element).toMatchSnapshot();
  });
});
