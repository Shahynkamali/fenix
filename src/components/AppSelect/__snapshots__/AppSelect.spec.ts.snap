// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSelect should accept a array of strings as a prop and set that as options 1`] = `
<div
  class="select"
>
  <label
    class="sr-only"
    for="select"
  >
    
    select
  
  </label>
   
  <select
    class="select__button"
    id="select"
  >
    <option
      value="Ascending Age "
    >
      
      Ascending Age 
    
    </option>
    <option
      value="Descending Age"
    >
      
      Descending Age
    
    </option>
  </select>
</div>
`;

exports[`AppSelect should display the selected option 1`] = `
<div
  class="select"
>
  <label
    class="sr-only"
    for="select"
  >
    
    select
  
  </label>
   
  <select
    class="select__button"
    id="select"
  >
    <option
      value="Ascending Age "
    >
      
      Ascending Age 
    
    </option>
    <option
      value="Descending Age"
    >
      
      Descending Age
    
    </option>
  </select>
</div>
`;
